lib/data_update_scripts/20220203194510_update_article_flag_without_parsing_article.rb

```diff
@@ -0,0 +1,12 @@
+module DataUpdateScripts
+  class UpdateArticleFlagWithoutParsingArticle
+    def run
+      # Query cribbed from https://dev.to/admin/blazer/queries/545-articles-containing-a-given-string-in-the-markdown?substring=cover_image%3A+
+      Article
+        .where.not(main_image: nil)
+        .where(main_image_from_frontmatter: false)
+        .where("body_markdown ILIKE ?", "---%cover_image:%---%")
```

Comment:
there appear to be some very small number of false positives in here (`cover_image:\n---` matches the pattern here) but some of those might have been because of the original bug that motivated this (cover_image: in frontmatter is removed and main_image remains)

I ran this as a spot check - most of them are either empty or some extra things (one had a leading paren, a few had  `cover_image:>- http...` which seems to have extracted correctly.

```sql
select id, title, body_markdown, main_image from articles
where main_image is not null
and main_image != ''
and main_image_from_frontmatter = false
and body_markdown like '---%cover_image:%---%'
and body_markdown !~ '---.*cover_image:[\s''"]*[h\/].*---';
```

Reply:
@djuber wouldn't you say that those false positives are fine though? if `cover_image` is detected it most likely means the article is using frontmatter.

Reply:
I went with an update.  The attribute is a somewhat self-correcting attribute.  Overtime, as folks edit things that we didn't get, it will be updated.